### PR TITLE
 feat: use flag library instead of custom flag parser 

### PR DESCRIPTION
**Pull request by rhnvrm**
Can be used to extend further as well, I am thinking of also adding custom shell flag so we can use zsh for example by passing a flag.

### CODE REVIEW

File: up.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"bufio"
 	"bytes"
 	"context"
+	"flag"
```
Comment (akavel) [inline up.go]: Could you please use [github.com/spf13/pflag](https://godoc.org/github.com/spf13/pflag) instead? I'm already starting to think of more flags in near future, and I think the POSIX approach will be more friendly to people in the long run.

File: up.go [hunk 2]
```diff
@@ -61,9 +62,28 @@ import (
...
 
+// Flag Settings
+var (
+	debugMode bool
+)
+
+func init() {
+	flag.BoolVar(&debugMode, "debug", false, "debug mode")
+	flag.BoolVar(&debugMode, "d", false, "debug mode (shorthand)")
+}
```
Comment (akavel) [inline up.go]: Please make it:

```
var (
    debugMode = pflag.Bool("debug", false, "...")
)
```

and adjust the code to use `*debugMode` appropriately.

I'm not sure yet if debug mode should be needed by people so often to already validate steal one of the potentially valuable one-letter shortcuts...

Thanks!

File: up.go [hunk 3]
```diff
@@ -163,14 +183,7 @@ func main() {
 }
 
 func parseFlags() {
```
Comment (akavel) [inline up.go]: Please delete the empty remains of the function. Thanks!

**Comment by rhnvrm** [comment]
Hey, thanks for the review, will push the changes in a while.

**Comment by akavel** [comment]
As to a custom flag for a replacement shell, it's totally something I'm interested in. Just not exactly sure how to design the interface, especially to take the final shebang into account. Please see #2 for some initial discussion; would you care to jump in with some thoughts and ideas?

File: up.go (hunk 1 above)
Comment (rhnvrm) [reply up.go]: I agree, we should stick with pflag

**Comment by rhnvrm** [comment]
@akavel I've pushed the changes.

I'll create another PR for the AUTHORS file as it does not exist as of now.

Also, will read #2 and jump in.

Thanks :smile:

File: up.go [hunk 4]
```diff
@@ -29,6 +29,8 @@ import (
 	"os/exec"
 	"sync"
 
+	flag "github.com/spf13/pflag"
```
Comment (akavel) [inline up.go]: Please don't rename pflag to flag. I understand they provide such an example in the docs, but the reasons they provide don't apply here, we can full well use the real name of the package.

Also please remove the empty line 33, there's no reason for this to stand out from other github dependencies.

File: up.go [hunk 5]
```diff
@@ -61,9 +63,23 @@ import (
 // TODO: [LATER] make it more friendly to infrequent Linux users by providing "descriptive" commands like "search" etc.
 // TODO: [LATER] advertise on: HN, r/programming, r/golang, r/commandline, r/linux, up-for-grabs.net; data exploration? data science?
 
+// Flag Settings
```
Comment (akavel) [inline up.go]: Please remove the comment, it doesn't contribute any significant information.

**Comment by akavel** [comment]
Thanks for your contribution!

**Comment by rhnvrm** [comment]
@akavel :+1:

**Pull request merged**